### PR TITLE
Feature example gotests

### PR DESCRIPTION
**Pull request by LakshmiMekala**
This PR provides end2end test sh files to go_tests for rules/examples so that it can be executed from any OS platform by running go test -v ./...

### CODE REVIEW

File: examples/flogo/simple-kafka/kakfa_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		fmt.Println("Error producer: ", err.Error())
+		os.Exit(1)
+	}
+
+	request := func() {
+		publish(`{"type":"grocery","totalPrice":"2001.0"}`, producer)
+	}
+	outpt := tests.CaptureOutput(request)
+
+	fmt.Println("********************", outpt, "888888888888888888888")
```
Comment (nareshkumarthota) [inline examples/flogo/simple-kafka/kakfa_test.go]: remove unwanted print statements

File: examples/flogo/simple-kafka/kakfa_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	assert.Equal(t, "success", result)
+	outpt = ""
+	result = ""
+
+}
+
+func TestSimpleKafkaJSON(t *testing.T) {
+	cmd := exec.Command("docker-compose")
+	err := cmd.Run()
+	if err == nil {
```
Comment (nareshkumarthota) [inline examples/flogo/simple-kafka/kakfa_test.go]: check this condition. I think this should be err != nil

File: examples/flogo/simple-kafka/kakfa_test.go (hunk 1 above)
Comment (LakshmiMekala) [reply examples/flogo/simple-kafka/kakfa_test.go]: Required changes were done

Comment (LakshmiMekala) [reply examples/flogo/simple-kafka/kakfa_test.go]: Required changes were done

File: examples/flogo/simple-kafka/kakfa_test.go (hunk 2 above)
Comment (LakshmiMekala) [reply examples/flogo/simple-kafka/kakfa_test.go]: Required changes were done

File: examples/flogo/creditcard/creditcard_test.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+		assert.Nil(t, err)
+		req.Header.Set("Content-Type", "application/json")
+		response, err := client.Do(req)
+		assert.Nil(t, err)
+		_, err = ioutil.ReadAll(response.Body)
+		assert.Nil(t, err)
+		response.Body.Close()
+		assert.Nil(t, err)
+	}
+	outpt = tests.CaptureOutput(request1)
```
Comment (rameshpolishetti) [inline examples/flogo/creditcard/creditcard_test.go]: @LakshmiMekala I feel like Capture**Std**Output() would be more appropriate name than CaptureOutput() as it is capturing standard output.

File: ruleapi/tests/common.go [hunk 4]
```diff
@@ -60,3 +69,66 @@ type txnCtx struct {
 	Testing *testing.T
 	TxnCnt  int
 }
+
+func CaptureOutput(f func()) string {
```
Comment (rameshpolishetti) [inline ruleapi/tests/common.go]: @LakshmiMekala I feel like CaptureStdOutput() would be more appropriate name than CaptureOutput() as it is capturing standard output.

File: examples/flogo/creditcard/creditcard_test.go (hunk 3 above)
Comment (LakshmiMekala) [reply examples/flogo/creditcard/creditcard_test.go]: required changes are done

File: ruleapi/tests/common.go (hunk 4 above)
Comment (LakshmiMekala) [reply ruleapi/tests/common.go]: required changes are done

**Comment by nareshkumarthota** [comment]
These changes merged into develop branch. Hence closing the PR.

**Pull request closed without merging**